packages/cuda/package.py

```diff
@@ -9,136 +9,13 @@
-        'Linux-x86_64': ('96863423feaa50b5c1c5e1b9ec537ef7ba77576a3986652351ae43e66bcd080c', 'https://developer.nvidia.com/compute/cuda/9.0/Prod/local_installers/cuda_9.0.176_384.81_linux-run')},
-    '8.0.61': {
-        'Linux-x86_64': ('9ceca9c2397f841024e03410bfd6eabfd72b384256fbed1c1e4834b5b0ce9dc4', 'https://developer.nvidia.com/compute/cuda/8.0/Prod2/local_installers/cuda_8.0.61_375.26_linux-run')},
-    '8.0.44': {
-        'Linux-x86_64': ('64dc4ab867261a0d690735c46d7cc9fc60d989da0d69dc04d1714e409cacbdf0', 'https://developer.nvidia.com/compute/cuda/8.0/prod/local_installers/cuda_8.0.44_linux-run')},
-    '7.5.18': {
-        'Linux-x86_64': ('08411d536741075131a1858a68615b8b73c51988e616e83b835e4632eea75eec', 'http://developer.download.nvidia.com/compute/cuda/7.5/Prod/local_installers/cuda_7.5.18_linux.run')},
-    '6.5.14': {
-        'Linux-x86_64': ('f3e527f34f317314fe8fcd8c85f10560729069298c0f73105ba89225db69da48', 'http://developer.download.nvidia.com/compute/cuda/6_5/rel/installers/cuda_6.5.14_linux_64.run')},
-}
-
-
-class Cuda(Package):
-    """CUDA is a parallel computing platform and programming model invented
-    by NVIDIA. It enables dramatic increases in computing performance by
-    harnessing the power of the graphics processing unit (GPU).
-
-    Note: This package does not currently install the drivers necessary
-    to run CUDA. These will need to be installed manually. See:
-    https://docs.nvidia.com/cuda/ for details."""
-
-    homepage = "https://developer.nvidia.com/cuda-zone"
-
-    for ver, packages in _versions.items():
-        key = "{0}-{1}".format(platform.system(), platform.machine())
-        pkg = packages.get(key)
-        if pkg:
-            version(ver, sha256=pkg[0], url=pkg[1], expand=False)
-
-    # macOS Mojave drops NVIDIA graphics card support -- official NVIDIA
-    # drivers do not exist for Mojave. See
-    # https://devtalk.nvidia.com/default/topic/1043070/announcements/faq-about-macos-10-14-mojave-nvidia-drivers/
-    # Note that a CUDA Toolkit installer does exist for macOS Mojave at
-    # https://developer.nvidia.com/compute/cuda/10.1/Prod1/local_installers/cuda_10.1.168_mac.dmg,
-    # but support for Mojave is dropped in later versions, and none of the
-    # macOS NVIDIA drivers at
-    # https://www.nvidia.com/en-us/drivers/cuda/mac-driver-archive/ mention
-    # Mojave support -- only macOS High Sierra 10.13 is supported.
-    conflicts('arch=darwin-mojave-x86_64')
-
-    depends_on('libxml2', when='@10.1.243:')
-
-    def setup_build_environment(self, env):
-        env.set('CUDAHOSTCXX', self.compiler.cxx)
-        if self.spec.satisfies('@10.1.243:'):
-            libxml2_home  = self.spec['libxml2'].prefix
-            env.set('LIBXML2HOME', libxml2_home)
-            env.append_path('LD_LIBRARY_PATH', libxml2_home.lib)
+class Cuda(Cuda):
 
     def setup_run_environment(self, env):
-        env.set('CUDA_HOME', self.prefix)
+       env.set('CUDA_HOME', self.prefix)
```

Comment:
I assume the missing space is simply an error.

We could call the super method here:

```python
def setup_run_environment(self, env):
    # let the original package setup its environment
    super().setup_run_environment(env)
    # add our own environment afterwards
    env.set(..., ...)
```

Reply:
This could also be buggy, depending on how spack uses inheritance for packages. If it's not, I think it's strictly better though.